types/tailwindcss/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for tailwindcss 2.0
+// Project: https://github.com/tailwindlabs/tailwindcss
+// Definitions by: Dolan Miu <https://github.com/dolanmiu>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.1
```

Comment:
BTW: this is the most recent, non preview version of TS, do we really want this for a package like
Tilewind CSS?

Reply:
The reason I made this change is because of this:

```sh
ERROR: 1865:13  expect  Compile error in typescript@4.0 but not in typescript@4.1.
Fix with a comment '// Minimum TypeScript Version: 4.1' just under the header.
Type '{ 0: string; 1: string; 2: string; 3: string; 4: string; 5: string; 6: string; 7: string; 8: string; 9: string; 10: string; 11: string; 12: string; 14: string; 16: string; 20: string; 24: string; 28: string; 32: string; ... 50 more ...; '-3.5': string; }' is not assignable to type 'TailwindStandardSpacing & TailwindStandardNegativeSpacing'.
```

Reply:
Many other libraries use 4.1, including `Google`:

```ts
// Type definitions for Google Geolocation 0.4.8
// Project: https://code.google.com/p/geo-location-javascript/
// Definitions by: Vincent Bortone <https://github.com/vbortone>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
// Minimum TypeScript Version: 4.1
```

Reply:
There's no reason to follow another library's precedent for minimum TypeScript version - either this package requires a particular version or it doesn't.  If @types/tailwindcss does require TS 4.1, then the header is correct, but it may still be undesirable.  Some type authors choose to avoid using features from newer versions of TS so that more consumers will be able to use the types.  For example, if someone were using TS 3.9 in their editor, they would be unable to use these types, because of the header.

Reply:
@amcasey The new type features will help simplify the type definitions a whole bunch, I used some of the newer features of typescript to make this type definition